openedx/core/djangoapps/content_libraries/signal_handlers.py

```diff
@@ -156,9 +148,8 @@ def library_collection_entity_saved(sender, instance, created, **kwargs):
     Sends a CONTENT_OBJECT_ASSOCIATIONS_CHANGED event for components added to a collection.
     """
     if created:
-        # Component.pk matches its entity.pk
-        component = get_component(instance.entity_id)
-        _library_collection_component_changed(component)
+        # FIXME: how to check if the instance is a component or container?
+        _library_collection_entity_changed(instance.entity)
```

Comment:
Hi @pomegranited!
I got stuck here.

I have a `CollectionPublishableEntity` from the `post_save` on `instance`. Ultimately, I need an OpaqueKey (`str`) to pass as the `object_id` parameter on the `CONTENT_OBJECT_ASSOCIATIONS_CHANGED` event, inside the `_library_collection_entity_changed` function.

I have the `entity_id` on the `PublishableEntity`, which translates to the `pk` for a `Component` OR a `Container`, but I don't know how to differentiate them.
The ideal would be to access the `OpaqueKey` directly from the `PublishableEntity`, but I don't think it is possible.

I appreciate some input on how to solve this.